### PR TITLE
Use key derivation for convergent nonce.

### LINKED ISSUE #1792: PBKDF convergent

### PR DESCRIPTION
**Pull request by jefferai**
Fixes #1792

### CODE REVIEW

File: builtin/logical/transit/policy.go [hunk 1]
```diff
@@ -365,8 +378,19 @@ func (p *Policy) Encrypt(context, nonce []byte, value string) (string, error) {
...
 
 	if p.ConvergentEncryption {
-		if len(nonce) != gcm.NonceSize() {
-			return "", errutil.UserError{Err: fmt.Sprintf("base64-decoded nonce must be %d bytes long when using convergent encryption with this key", gcm.NonceSize())}
+		if p.ConvergentVersion == 1 {
+			if len(nonce) != gcm.NonceSize() {
+				return "", errutil.UserError{Err: fmt.Sprintf("base64-decoded nonce must be %d bytes long when using convergent encryption with this key", gcm.NonceSize())}
+			}
+		} else {
+			if len(nonce) < 12 {
```
Comment (vishalnayak) [inline builtin/logical/transit/policy.go]: Instead of 12, how about we use `gcm.NonceSize()` here?

File: builtin/logical/transit/path_keys.go [hunk 2]
```diff
@@ -109,6 +109,9 @@ func (b *backend) pathPolicyRead(
 	if p.Derived {
 		resp.Data["kdf_mode"] = p.KDFMode
 		resp.Data["convergent_encryption"] = p.ConvergentEncryption
+		if p.ConvergentEncryption {
```
Comment (vishalnayak) [inline builtin/logical/transit/path_keys.go]: Can we remove this `if` check? I guess it'll always be set considering that we upgrade the entry while getting the policy.

**Comment by vishalnayak** [comment]
Two minor comments, otherwise LGTM!

File: builtin/logical/transit/path_keys.go (hunk 2 above)
Comment (jefferai) [inline builtin/logical/transit/path_keys.go]: I'll always be set, but no point in outputting it to the user if convergent encryption isn't turned on.

**Pull request merged**